6_Agent_Deployment/backend_agent_api/agent_api.py

```
@@ -493,6 +493,133 @@
+    try:
+        # Enforce per-user path ownership (defense-in-depth)
+        user_id = user.get("id")
+        if not user_id or not request.file_path.startswith(f"{user_id}/"):
+            raise HTTPException(status_code=403, detail="Forbidden: file path not owned by user")
+        
+        # Download file from Supabase storage (sync call -> offload)
+        file_bytes = await asyncio.to_thread(
+            lambda: supabase.storage.from_("documents").download(request.file_path)
+        )
+        if not file_bytes:
+            raise HTTPException(status_code=404, detail="File not found in storage")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)